include/keycombination.h

```diff
@@ -0,0 +1,34 @@
+#ifndef NEWSBOAT_KEYCOMBINATION_H_
+#define NEWSBOAT_KEYCOMBINATION_H_
+
+#include <ostream>
+#include <string>
+
+namespace newsboat {
+
+class KeyCombination {
+public:
+	explicit KeyCombination(const std::string& key, bool shift = false, bool control = false,
+		bool alt = false);
```

Comment:
I just want to point out that these three booleans can be mixed up easily as there is no natural order for <kbd>Shift</kbd>, <kbd>Ctrl</kbd> and <kbd>Alt</kbd> in my opinion. Any order is equally valid.

However, it might be quite some overkill to introduce typed enums for these modifiers, e.g. `NoShift`/`Shift` etc. A single combinable modifier type would be probably even better, there's a reason, that lots of APIs do it like that.

Anyways, maybe not so bad as this constructor is only used at a few places. Just tests might still benefit from dedicated types.

Reply:
I've changed these into separate enums.
Not sure if it is easer to read, but it is easier to avoid accidental issues in reordering the arguments.

Reply:
Yes, the `FooState::` clutters the code quite a bit, that's right. But in the end it is hopefully faster to read through that lengthy code instead of jumping back to the definition all the time to see the argument order. *Maybe* in tests some `using enum …` helps in that regard, not sure, I don't consider myself a real C++ programmer.

In `to_bindkey_string()` the use of the boolean modifier helper methods *might* make the code slightly shorter and hence a tad more readable. But that's the only other thing I can think of.